### PR TITLE
Group by message and add contextual message to report

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes https://github.com/gradle/gradle-private/issues/4455

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/ResolutionFailureHandler.java [hunk 1]
```diff
@@ -243,6 +243,7 @@ private AbstractResolutionFailureException reportExceptionAsProblem(AbstractReso
             ResolutionFailureProblemId problemId = exception.getFailure().getProblemId();
             builder.id(TextUtil.screamingSnakeToKebabCase(problemId.name()), problemId.getDisplayName(), GradleCoreProblemGroup.variantResolution())
                 .contextualLabel(exception.getMessage())
+                .details(exception.getFailure().describeRequestTarget())
```
Comment (donat) [inline platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/ResolutionFailureHandler.java:246]: @gradle/bt-jvm fyi

File: platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt [hunk 2]
```diff
@@ -127,9 +125,12 @@ class JsonProblemWriter(private val problem: Problem, private val failureDecorat
                         )
                     }
                 }
+                problem.contextualLabel?.let {
+                    property("contextualLabel", it)
+                }
                 problem.definition.documentationLink?.let { property("documentationLink", it.url) }
                 problem.exception?.let { writeError(failureDecorator.decorate(failureFactory.create(it))) }
-                property("category") {
+                property("group") {
```
Comment (donat) [inline platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt:133]: ❓Shouldn't we group by ID? ie. the property name should be ID and the list should contain `DefaultProblemId`s.

**Review by donat — CHANGES_REQUESTED** [review]
I've added a few items.
Please make sure we have an issue explaining the problem and link it to the PR.

File: platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/ResolutionFailureHandler.java (hunk 1 above)
Comment (tresat) [reply platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/ResolutionFailureHandler.java:246]: ❗ I don't think you want to use this.  The target can vary in both type and format based on the failure type.

The javadoc on `details` says "The long description of this problem."  I think. `exception.getFailure().getProblemId().getDisplayName()` might be more what you're looking for.

**Review by tresat — CHANGES_REQUESTED** [review]

File: platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt (hunk 2 above)
Comment (reinsch82) [reply platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt:133]: this refers to the same thing we decided to call "group" [here](https://github.com/gradle/configuration-cache-report/pull/26#discussion_r1745115032)

**Review by donat — APPROVED** [review]

**Pull request merged**